include/gemmini.h

```diff
@@ -797,17 +803,20 @@ static void matmul_cpu(size_t DIM_I, size_t DIM_J, size_t DIM_K,
   } else {
     for (size_t i = 0; i < DIM_I; i++) {
       for (size_t j = 0; j < DIM_J; j++) {
+        const elem_t* a = !transA ? (A + (i * stride_A)) : A + i;
+        const elem_t* b = !transB ? (B + j) : (B + (j * stride_B));
```

Comment:
I would rather calculate these lines directly in the k-for loop, like this:

```
        for (size_t k = 0; k < DIM_K; k++) {
          const elem_t* a = !transA ? (A + (i * stride_A) + k) : (A + (k * stride_A) + i);
         const elem_t* b = !transB ? (B + (k * stride_B) + j) : (B + (j * stride_B) + k);

          sum += (GEMMINI_SCALE(*a, A_scale_factor) * GEMMINI_SCALE(*b, B_scale_factor));
        }
```

That would reduce the amount of state in our design.

Reply:
Done. I originally had it outside the k-for loop to avoid the performance penalty from branching too much  (due to the ternary) within the innermost loop though. Although I suppose if you really wanted to maximize performance it'd be best to unroll the cases for the 4 combinations of `transA` and `transB` and have separate loops for each. Given that the cpu code is used as reference for benchmarks, should I switch to doing that?

Reply:
Discussed offline – decided we can keep the computation outside the k-loop for now to avoid performance penalty, but not worth unrolling for the four cases since that'd make the code too ugly.